oidc-wp/openid-connect-generic
# feat(Sessions): Add remember me feature via a filter

### All Submissions:






### Changes proposed in this Pull Request:



Closes #512 .

This change allows the user to, using a filter, login users via the rememberme option in WordPress.
The wordpress login cookies will then be set for 14 days by default instead of two days as a session.

I originally wanted to use the token expiration date to do so, however this doesn't seem possible as you cannot pass an expiration date to the wp_set_auth_cookie function. So instead I chose to implement the remember me feature as used by WordPress itself.

What could maybe be possible is to dynamically add a filter for auth_cookie_expiration, so the WP user cookie expiration can be set to the same value. This filter needs to be added temporarily, and then removed.
However this is kind of hacky and i think it would be better to submit that as a separate PR after this one is approved.

### How to test the changes in this Pull Request:

1. In your theme, add `add_filter( 'openid-connect-generic-remember-me', '__return_true' );` to the functions.php file.
2. Login using the openid connect plugin
3. Open the development tools of your browser, and look at the cookie expiration. It should not be a session cookie, and it should be valid for 2 weeks.
![image](https://github.com/oidc-wp/openid-connect-generic/assets/50165380/dc730bdd-782a-4cb3-8755-f6182c61bcba)


### Other information:




### Changelog entry

> Added the `openid-connect-generic-remember-me` filter to allow theme makers to login users using the remember me option.
